### PR TITLE
Fixed bug in field indexes related to pushdown of indexes in subqueries

### PR DESCRIPTION
**Pull request by zachmu**

### CODE REVIEW

File: sql/analyzer/indexed_joins.go [hunk 1]
```diff
@@ -173,29 +174,29 @@ func replaceTableAccessWithIndexedAccess(
 		}
 
 		// the condition's field indexes might need adjusting if the order of tables changed
-		cond, err := FixFieldIndexes(scope, append(schema, append(left.Schema(), right.Schema()...)...), node.Cond)
+		cond, err := FixFieldIndexes(scope, nil, append(schema, append(left.Schema(), right.Schema()...)...), node.Cond)
```
Comment (andrew-wm-arthur) [inline sql/analyzer/indexed_joins.go:177]: intended `nil`?

File: sql/analyzer/fix_field_indexes.go [hunk 2]
```diff
@@ -63,13 +68,18 @@ func FixFieldIndexes(scope *Scope, schema sql.Schema, exp sql.Expression) (sql.E
...
-						return expression.NewGetFieldWithTable(
-							scopeLen-offset+i,
-							e.Type(),
-							e.Table(),
-							e.Name(),
-							e.IsNullable(),
-						), nil
+						newIndex := scopeLen - offset + i
+						if e.Index() != newIndex {
+							a.Log("Rewriting field %s.%s from index %d to %d", e.Table(), e.Name(), e.Index(), newIndex)
```
Comment (andrew-wm-arthur) [inline sql/analyzer/fix_field_indexes.go:73]: seems like a lot of threading to get the log statement here

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM, seems worth revisiting as the column index mappings are gathering complexity

File: sql/analyzer/fix_field_indexes.go (hunk 2 above)
Comment (zachmu) [reply sql/analyzer/fix_field_indexes.go:73]: Yeah, eventually we need to put the logger in the session or something

File: sql/analyzer/indexed_joins.go (hunk 1 above)
Comment (zachmu) [reply sql/analyzer/indexed_joins.go:177]: Nope, good catch

**Pull request merged**